addons/test_mail/tests/test_mail_composer.py

```diff
@@ -1852,6 +1859,79 @@ def test_mail_composer_delete(self):
         self.assertEqual(len(self._new_msgs), 2, 'Should have created 1 mail.mail per record')
         self.assertFalse(self._new_msgs.exists(), 'Should have deleted mail.message records')
 
+    @users('employee')
+    @mute_logger('odoo.tests', 'odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    def test_mail_composer_duplicates(self):
+        """ Ensures emails sent to the same recipient multiple times
+            are only sent when they are not duplicates
+        """
+        self.template.write({
+            'auto_delete': False,
+            'body_html': '<p>Common Body</p>',
+            'email_to': 'test@test.lan',
+            'subject': 'Common Subject',
+            'partner_to': '',
+        })
+
+        # Guarantee points of variation for test_report_3
+        self.test_records[0].write({'name': 'A', 'count': 1})
```

Comment:
Would probably play with record emails values (continuing previous comment), indicating who are the same and who are not.

Reply:
With specific test, like: one record has ``email.duplicate@test.example.com`` email, and another has ``"Formatted Duplicate" <email.duplicate@test.example.com>`` and see how it behaves. Tde, good ideas since 2011.

Reply:
Yep good idea